stl/inc/span

```diff
@@ -0,0 +1,723 @@
+// span standard header
+
+// Copyright (c) Microsoft Corporation.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#pragma once
+#ifndef _SPAN_
+#define _SPAN_
+#include <yvals_core.h>
+#if _STL_COMPILER_PREPROCESSOR
+
+#if _HAS_CXX20
+#include <array>
+#include <cstddef>
+#include <type_traits>
+
+#pragma pack(push, _CRT_PACKING)
+#pragma warning(push, _STL_WARNING_LEVEL)
+#pragma warning(disable : _STL_DISABLED_WARNINGS)
+_STL_DISABLE_CLANG_WARNINGS
+#pragma push_macro("new")
+#undef new
+
+_STD_BEGIN
+
+inline constexpr size_t dynamic_extent = static_cast<size_t>(-1);
+
+// CLASS TEMPLATE _Span_extent_type
+template <size_t _Extent>
+class _Span_extent_type {
+public:
+    constexpr _Span_extent_type() noexcept = default;
+
+    constexpr explicit _Span_extent_type([[maybe_unused]] const size_t _Size) noexcept {
+#if _CONTAINER_DEBUG_LEVEL > 0
+        _STL_VERIFY(_Size == _Extent, "Mismatch between static extent and size of initializing data");
+#endif // _CONTAINER_DEBUG_LEVEL > 0
+    }
+
+    template <size_t _OtherExtent>
+    constexpr explicit _Span_extent_type(const _Span_extent_type<_OtherExtent> _Ext) noexcept {
+        static_assert(_OtherExtent == _Extent || _OtherExtent == dynamic_extent,
+            "Mismatch between static extent and size of initializing data.");
+#if _CONTAINER_DEBUG_LEVEL > 0
+        if constexpr (_OtherExtent == dynamic_extent) {
+            _STL_VERIFY(_Ext._Size() == _Extent, "Mismatch between static extent and size of initializing data");
+        }
+#endif // _CONTAINER_DEBUG_LEVEL > 0
+    }
+
+    _NODISCARD constexpr size_t _Size() const noexcept {
+        return _Extent;
+    }
+};
+
+template <>
+class _Span_extent_type<dynamic_extent> {
+public:
+    constexpr _Span_extent_type() noexcept = default;
+
+    constexpr explicit _Span_extent_type(const size_t _Size) noexcept : _Mysize(_Size) {}
+
+    template <size_t _OtherExtent>
+    constexpr explicit _Span_extent_type(const _Span_extent_type<_OtherExtent> _Ext) noexcept : _Mysize(_Ext._Size()) {}
+
+    _NODISCARD constexpr size_t _Size() const noexcept {
+        return _Mysize;
+    }
+
+private:
+    size_t _Mysize{0};
+};
+
+// CLASS TEMPLATE _Span_storage
+template <class _Ptr, size_t _Extent>
+class _Span_storage : public _Span_extent_type<_Extent> {
+    using _Mybase = _Span_extent_type<_Extent>;
+
+public:
+    constexpr _Span_storage() = default;
+
+    explicit constexpr _Span_storage(const size_t _Ext, const _Ptr _Data) : _Mybase(_Ext), _Mydata(_Data) {}

```

Comment:
You are now "shadowing" the member function `_Data` with the constructor parameter `_Data`; similarly in the next constructor. Our convention is to avoid shadowing, so any other name should be chosen. `_Data_` would be acceptable (we use this pattern for other constructor parameters).

Reply:
Thanks a lot. I thought about better names and gave up